pkg/gofr/datasource/kv-store/badger/badger.go

```diff
@@ -0,0 +1,103 @@
+package badger
+
+import (
+	"context"
+	"github.com/dgraph-io/badger/v4"
+)
+
+type Configs struct {
+	DirPath string
+}
+
+type client struct {
+	db      *badger.DB
+	configs Configs
+	logger  Logger
+	metrics Metrics
+}
+
+func New(configs Configs) *client {
+	return &client{configs: configs}
+}
+
+// UseLogger sets the logger for the MongoDB client which asserts the Logger interface.
+func (c *client) UseLogger(logger interface{}) {
```

Comment:
Why not using this?


```suggestion
func (c *client) UseLogger(logger Logger) {
```

Is it for mock consideration?

Reply:
provider interface has interface, that is why we are accepting interfaces.
Else we would need to have a common source for the Logger and Metrics interface which would bring dependency, which is we are refraining from

Reply:
OK. So maybe here you could define an interface with one method that will be implemented by each Logger (and do the same for Metrics)

There is usually one thing that can be shared.

This way the code is clearer. IDE are able to autocomplete.

But I concede, if it won't change the type assertion you are doing there. But it would be better I think.

Reply:
It can be part of a bigger refactor, as all external DBs need to be refactored

Reply:
Of course, it has to be a separate PR. You would have to rewrite a lot of files. It's out of the scope of this PR

Reply:
I opened 

- https://github.com/gofr-dev/gofr/issues/840